app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,93 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.from_omniauth(request.env['omniauth.auth'])
+    session[:user_id] = user.id
+
+    redirect_to(
+      admin_path,
+      notice: t('omniauth_callbacks.success')
+    )
+  end
+
+  def destroy
+    if params[:SAMLRequest]
+      idp_logout_request
+    elsif params[:SAMLResponse]
+      validate_slo_response
+    else
+      sp_logout_request
+    end
+  end
+
+  def setup
+    if params.key?(:loa)
+      request.env['omniauth.strategy'].options[:authn_context] = [
+        "http://idmanagement.gov/ns/assurance/loa/#{params[:loa]}",
+        'http://idmanagement.gov/ns/requested_attributes?ReqAttr=email,phone,first_name,last_name,ssn'
```

Comment:
do you need SSN for this app? Don't ask for it unless you have a business need, because that's PII.